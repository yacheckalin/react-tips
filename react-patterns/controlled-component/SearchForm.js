import React from "react";

class SearchForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = { searchField: "" };
  }
  handleSearch = (e) => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  render() {
    return (
      <div className="row">
        <form className="col s12">
          <div className="row">
            <div className="input-field col s6">
              <input
                id="search_prefix"
                type="text"
                name="searchField"
                className="input-field"
                value={this.state.searchField}
                onChange={this.handleSearch}
              />
              <label htmlFor="search_prefix"></label>
            </div>
          </div>
        </form>
      </div>
    );
  }
}

export default SearchForm;
