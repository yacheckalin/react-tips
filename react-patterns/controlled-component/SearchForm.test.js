import React from "react";
import toJSON from "enzyme-to-json";
import SearchForm from "./SearchForm";
import { shallow } from "enzyme";

describe("<SearchForm />", () => {
  it("renders and displays properly", () => {
    const wrapper = shallow(<SearchForm />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
