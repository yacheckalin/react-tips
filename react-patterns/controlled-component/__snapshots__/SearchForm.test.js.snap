// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> renders and displays properly 1`] = `
<div
  className="row"
>
  <form
    className="col s12"
  >
    <div
      className="row"
    >
      <div
        className="input-field col s6"
      >
        <input
          className="input-field"
          id="search_prefix"
          name="searchField"
          onChange={[Function]}
          type="text"
          value=""
        />
        <label
          htmlFor="search_prefix"
        />
      </div>
    </div>
  </form>
</div>
`;
